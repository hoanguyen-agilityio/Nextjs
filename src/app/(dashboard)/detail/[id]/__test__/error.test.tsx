import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ErrorDetailPage from '../error';

describe('ErrorDetailPage Component', () => {
  test('renders ErrorDetailPage correctly', () => {
    const mockReset = jest.fn();
    const mockError = {
      message: 'An unexpected error occurred.',
      name: 'Error',
      stack: 'Error stack trace',
      digest: 'unique-digest',
    };

    const { asFragment } = render(
      <ErrorDetailPage error={mockError} reset={mockReset} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
