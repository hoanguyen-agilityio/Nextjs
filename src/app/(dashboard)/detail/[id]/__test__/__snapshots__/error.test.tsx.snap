// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDetailPage Component renders ErrorDetailPage correctly 1`] = `
<DocumentFragment>
  <section
    class="min-h-screen flex justify-center"
  >
    <div
      class="max-w-2xl p-5 flex flex-col items-center justify-center m-auto"
    >
      <h1
        class="text-3xl font-bold text-red-500 mb-4 text-center"
      >
        Error 500: Internal server error
      </h1>
      <p
        class="text-red-600 text-xl"
      >
        An unexpected error occurred.
      </p>
      <button
        class="mt-4 inline-block px-4 py-2 bg-ratio-400 text-white rounded-md dark:border-blue-500 text-xl"
      >
        Reset
      </button>
    </div>
  </section>
</DocumentFragment>
`;
